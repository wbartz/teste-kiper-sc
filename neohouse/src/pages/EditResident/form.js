import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { Controller } from 'react-hook-form';
import { formatCpf, formatPhone, formatDate } from '../../helpers';
import TextField from '../../components/TextField';
import SelectField from '../../components/SelectField';

const Form = ({ errors, control }) => {
  const [fullName, setFullName] = useState();
  const [email, setEmail] = useState();
  const [cpf, setCpf] = useState();
  const [phone, setPhone] = useState();
  const [birthday, setBirthday] = useState();
  const [accountable, setAccountable] = useState(0);
  const statusOptions = [
    { value: false, label: 'Não' },
    { value: true, label: 'Sim' },
  ];

  return (
    <>
      <div className="row">
        <div className="col m4">
          <Controller
            as={
              <SelectField
                value={accountable.toString()}
                options={statusOptions}
                onChange={setAccountable}
              />
            }
            name="accountable"
            label="Responsável"
            control={control}
          />
        </div>
      </div>
      <div className="row">
        <div className="col m8">
          <Controller
            as={<TextField onChange={setFullName} />}
            valueName={fullName}
            name="full_name"
            label="Nome completo"
            defaultValue={fullName}
            control={control}
            error={errors.fullName && errors.fullName.message}
            size="auto"
          />
        </div>
      </div>
      <div className="row">
        <div className="col m4">
          <Controller
            as={<TextField onChange={setBirthday} />}
            valueName={birthday}
            name="birthday"
            label="Data de nascimento"
            defaultValue={birthday}
            type="datepicker"
            control={control}
            format={formatDate}
            error={errors.birthday && errors.birthday.message}
            size="auto"
          />
        </div>
        <div className="col m4">
          <Controller
            as={<TextField onChange={setCpf} />}
            valueName={cpf}
            name="cpf"
            label="CPF"
            defaultValue={cpf}
            format={formatCpf}
            control={control}
            error={errors.cpf && errors.cpf.message}
            size="auto"
          />
        </div>
      </div>
      <div className="row">
        <div className="col m4">
          <Controller
            as={<TextField onChange={setEmail} />}
            valueName={email}
            name="email"
            label="E-mail"
            defaultValue={email}
            control={control}
            error={errors.email && errors.email.message}
            size="auto"
          />
        </div>
        <div className="col m4">
          <Controller
            as={<TextField onChange={setPhone} />}
            valueName={phone}
            name="phone"
            label="Telefone"
            defaultValue={phone}
            format={formatPhone}
            control={control}
            error={errors.phone && errors.phone.message}
            size="auto"
          />
        </div>
      </div>
    </>
  );
};

Form.propTypes = {
  errors: PropTypes.oneOfType([PropTypes.object]),
  control: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

Form.defaultProps = {
  errors: {},
};

export default Form;
