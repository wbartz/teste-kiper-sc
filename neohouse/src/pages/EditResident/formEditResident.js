import M from 'materialize-css';
import PropTypes from 'prop-types';
import React, { useCallback, useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { LOG } from '../../helpers';
import Form from './form';
import './index.scss';
import Success from './success';
import validationSchema from './validations';

const FormEditResident = ({ history, location, editResident, getResident }) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [hasData, setData] = useState(false);
  const { handleSubmit, control, errors, setValue } = useForm({
    validationSchema,
  });
  LOG(location.state);
  const { resident_id } = location.state;

  const setValues = useCallback(
    (info) => {
      Object.keys(info).map((key) => {
        let value = info[key];
        setValue(key, value);
        M.updateTextFields();
        return true;
      });
    },
    [setValue]
  );

  const getDate = useCallback(async () => {
    await getResident(resident_id, setValues);
  }, [getResident, resident_id, setValues]);

  useEffect(() => {
    if (!hasData) {
      getDate();
      setData(true);
    }
  }, [location, getDate, hasData]);

  const onSuccess = () => {
    setShowSuccess(true);
  };

  const handleAdd = (values) => {
    editResident(resident_id, { ...values }, onSuccess);
  };

  return (
    <div className="page edit-resident-page">
      <form onSubmit={handleSubmit(handleAdd)}>
        <Form errors={errors} control={control} />
        <div className="row buttons">
          <div className="col m8">
            <div className="col m2 lef">
              <button
                type="button"
                className="btn btn-flat"
                onClick={() => history.goBack()}
              >
                Cancelar
              </button>
            </div>
            <div className="col m2 right">
              <button
                type="submit"
                className="btn btn-submit primary"
                disabled={errors.length > 0}
              >
                Salvar
              </button>
            </div>
          </div>
        </div>
      </form>

      <Success
        show={showSuccess}
        onClose={() => {
          setShowSuccess(false);
        }}
      />
    </div>
  );
};

FormEditResident.propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  location: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  editResident: PropTypes.func.isRequired,
  getResident: PropTypes.func.isRequired,
};

export default FormEditResident;
