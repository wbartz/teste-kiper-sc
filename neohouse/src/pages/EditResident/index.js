import React, { useContext } from 'react';
import { withRouter } from 'react-router-dom';
import { AppContext } from '../../containers/StoreProvider';
import FormEditResident from './formEditResident';

export default withRouter((props) => (
  <FormEditResident {...useContext(AppContext)} {...props} />
));
