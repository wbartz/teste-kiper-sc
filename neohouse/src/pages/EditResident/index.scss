.edit-resident-page {
  .row.buttons {
    .col {
      padding: 0;
    }
  }
}