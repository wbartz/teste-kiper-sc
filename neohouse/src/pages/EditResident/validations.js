import * as yup from 'yup';

const phoneRegEx = /\(\d{2,}\) \d{1,} \d{4,} \d{4}/g;

const defaulSchema = yup.object().shape({
  email: yup
    .string()
    .required('Este campo é obrigatório')
    .email('O e-mail informado não é válido'),
  full_name: yup.string().required('Este campo é obrigatório'),
  cpf: yup.string().required('Este campo é obrigatório').min(14),
  birthday: yup.string().required('Este campo é obrigatório'),
  phone: yup.string().matches(phoneRegEx, {
    message: 'Número de telefone inválido',
    excludeEmptyString: true,
  }),
});

export default defaulSchema;
