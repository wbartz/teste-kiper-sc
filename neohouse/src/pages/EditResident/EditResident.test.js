import { shallow } from 'enzyme';
import React from 'react';
import EditResident from './formEditResident';

describe('EditResident', () => {
  let wrapper;
  const props = {
    history: {
      push: jest.fn(),
    },
    location: {
      state: {
        block_id: 1,
        apartment_id: 1,
      }
    },
    getResident: jest.fn(),
    editResident: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<EditResident {...props} />);
  });

  it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
