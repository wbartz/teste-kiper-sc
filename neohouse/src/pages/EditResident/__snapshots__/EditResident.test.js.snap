// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditResident render without crashing 1`] = `ShallowWrapper {}`;
